pkg/apis/ocmagent/v1alpha1/ocmagent_types.go

```diff
@@ -4,17 +4,22 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type AgentConfig struct {
+	// OcmBaseUrl defines the OCM api endpoint for OCM agent to access
+	OcmBaseUrl string `json:"ocmBaseUrl"`
+
+	// Services defines the supported OCM services, eg, service_log, cluster_management
+	Services []string `json:"services"`
+}
+
 // OcmAgentSpec defines the desired state of OcmAgent
 type OcmAgentSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	// OcmBaseUrl defines the OCM api endpoint for OCM agent to access
-	OcmBaseUrl string `json:"ocmBaseUrl"`
-
-	// Services defines the supported OCM services, eg, service_log, cluster_management
-	Services []string `json:"services"`
+	// AgentConfig refers to OCM agent config fields separated
+	Config AgentConfig `json:"agentConfig"`
```

Comment:
Maybe call this `AgentConfig` to keep it consistent with naming of the json field?

Reply:
Thanks for the review, made the change as per the suggestion.